src/lib.ts

```diff
@@ -485,18 +488,65 @@ export function userConfirmDialog(msg: string, defaultValue: boolean, timeOut?:
   return userConfirm(msg);
 }
 
-export const latteActionSourceFinderConstraints = {
-  allowedAction: (action: ActionSource): boolean => {
-    if (!have($item`latte lovers member's mug`)) return true;
-    const forceEquipsOtherThanLatte = (
-      action?.constraints?.equipmentRequirements?.().maximizeOptions.forceEquip ?? []
-    ).filter((equipment) => equipment !== $item`latte lovers member's mug`);
-    return (
-      forceEquipsOtherThanLatte.every((equipment) => toSlot(equipment) !== $slot`off-hand`) &&
-      sum(forceEquipsOtherThanLatte, weaponHands) < 2
-    );
-  },
-};
+function determineFreeBunnyBanish(): boolean {
+  const expectedPocketProfFights = !have($familiar`Pocket Professor`)
+    ? 0
+    : (!get("_garbo_meatChain", false) ? Math.max(10 - get("_pocketProfessorLectures"), 0) : 0) +
+      (!get("_garbo_weightChain", false) ? Math.min(15 - get("_pocketProfessorLectures"), 5) : 0);
+  const expectedDigitizesDuringGregs =
+    SourceTerminal.have() && get("_sourceTerminalDigitizeUses") < 3 ? 3 : 0; // To encounter 3 digitize monsters it takes 91 adventures. Just estimate we fight all 3 to be safe.
```

Comment:
Yeah that is fine, we can do a more precise estimate but it's better to err on the side of caution